src/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,268 @@
+import React, {
+  FC,
+  ReactNode,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react'
+import styled, { css } from 'styled-components'
+import { Text } from '../Text'
+import { theme } from '../theme'
+
+type Position = 'top' | 'bottom' | 'left' | 'right'
+type Size = 'small' | 'large'
+type ArrowPosition = 'left' | 'right' | 'center' | 'top' | 'bottom'
+
+export interface TooltipProps {
+  children: ReactNode
+  title?: string
+  content: string | ReactNode
+  position: Position
+  underline?: boolean
+  size?: Size
+  arrowPosition?: ArrowPosition
```

Comment:
considering that we dynamically set the tooltip position based on the element, should we also derive the arrow position as well and remove both `position` and `arrowPosition` as props?
Just trying to avoid conflicting behaviours between default values and resizes. 
If we agree with design that the default position is always top, then I guess everything else will change based on scrolling and resize. 
I know flexibility is a good thing but sometimes too much might not be? What do you think?

Reply:
@fleonard I've taken some time to have a look and I _think_ I've come up with a decent solution. From the Figma file the arrow position looks like it can (and should) be placed anywhere so I've left this to be used defineable as a prop (the default will still be `center`) and removed the `position` prop as we're defaulting the tooltip to always display at the top.

The Tooltip now checks the page bounds and makes sure it's in view, defaulting to top and moving around the content if out of bounds. I've also made it default back to top when the space is available. 

I think this allows enough flexibility with the arrow position whilst not being too complex in implementation due to removing the tooltip `position` prop. Let me know what you think!